api/producer/firehose/tests/test_producer_firehose_index.py

```diff
@@ -0,0 +1,102 @@
+"""
+This file contains one function testing three cases in a single pytest.
+The reason for this setup is that the Firehose cycle
+(event queueing, lambda execution, firehose output) can take 5-7 minutes
+and so all test cases are submitted in advance of the test via the fixtures:
+
+    _submit_good_cloudwatch_data,
+    _submit_bad_cloudwatch_data,
+    _submit_very_bad_cloudwatch_data
+
+The S3 output file prefix and keys can't be known exactly, even at runtime, and
+so the relevant parts of S3 must be trawled for output files which appear after a
+certain point in time.
+"""
+
+import base64
+import gzip
+import json
+from datetime import datetime, timedelta
+
+import pytest
+
+from api.producer.firehose.tests.e2e_utils import (
+    MAX_RUNTIME,
+    parse_prefix,
+    retrieve_firehose_output,
+)
+
+
+@pytest.mark.slow
+@pytest.mark.integration
+@pytest.mark.firehose
+def test_firehose_output(
+    session,
+    bucket_name,
+    prefix_template,
+    error_prefix_template,
+    global_event_handler,
```

Comment:
I think im being dense but how does this input work from the conftest file? is it because the value was set to pytest session?

Reply:
when pytest runs any test, it looks for the arguments from matching fields in `parametrize` and from matching fields from any available `fixture`. Fixtures are in scope if they're in the same module, or a `conftest.py` in the same directory as the test, or the same directory from where `pytest` is called